tests/test_sequence_sampler.py

```diff
@@ -137,7 +135,48 @@ def test_modulation(mod_seq: pulser.Sequence) -> None:
             getattr(input_ch_samples.modulate(chan), qty),
             getattr(output_ch_samples, qty),
         )
-    assert input_ch_samples.modulate(chan).slots == output_ch_samples.slots
+
+
+def test_modulation_local(mod_device):
+    seq = pulser.Sequence(pulser.Register.square(2), mod_device)
+    seq.declare_channel("ch0", "rydberg_local", initial_target=0)
+    ch_obj = seq.declared_channels["ch0"]
+    pulse1 = Pulse.ConstantPulse(500, 1, -1, 0)
+    pulse2 = Pulse.ConstantPulse(200, 2.5, 0, 0)
+    partial_fall = pulse1.fall_time(ch_obj) // 3
+    seq.add(pulse1, "ch0")
+    seq.delay(partial_fall, "ch0")
+    seq.add(pulse2, "ch0")
+    seq.target(1, "ch0")
+    seq.add(pulse1, "ch0")
+
+    input_samples = sample(seq)
+    output_samples = sample(seq, modulation=True)
+    assert input_samples.max_duration == seq.get_duration()
+    assert output_samples.max_duration == seq.get_duration(
+        include_fall_time=True
+    )
+
+    # Check that the target slots account for fall time
+    in_ch_samples = input_samples.channel_samples["ch0"]
+    out_ch_samples = output_samples.channel_samples["ch0"]
+    expected_slots = deepcopy(in_ch_samples.slots)
+    # The first slot should extend to the second
+    expected_slots[0].tf += partial_fall
+    assert expected_slots[0].tf == expected_slots[1].ti
+    # The next slots should fully account for fall time
+    expected_slots[1].tf += pulse2.fall_time(ch_obj)
+    expected_slots[2].tf += pulse1.fall_time(ch_obj)
+
+    assert out_ch_samples.slots == expected_slots
```

Comment:
I have not idea of what going on.

Reply:
Is it clearer after the explanation above?

Reply:
Yes, much clearer!